File: sportswire/lib/core/team_config.dart
import 'package:flutter/material.dart';

class TeamData {
  final String fullName;
  final Color primaryColor;
  final Color accentColor;

  const TeamData({
    required this.fullName,
    required this.primaryColor,
    required this.accentColor,
  });
}

class TeamConfig {
  static const Map<String, TeamData> nba = {
    'ATL': TeamData(fullName: 'Atlanta Hawks',          primaryColor: Color(0xFFC1052A), accentColor: Color(0xFFFDB927)),
    'BOS': TeamData(fullName: 'Boston Celtics',         primaryColor: Color(0xFF007A33), accentColor: Color(0xFFBA9653)),
    'BKN': TeamData(fullName: 'Brooklyn Nets',          primaryColor: Color(0xFF000000), accentColor: Color(0xFFFFFFFF)),
    'CHA': TeamData(fullName: 'Charlotte Hornets',      primaryColor: Color(0xFF1D1160), accentColor: Color(0xFF00788C)),
    'CHI': TeamData(fullName: 'Chicago Bulls',          primaryColor: Color(0xFFCE1141), accentColor: Color(0xFF000000)),
    'CLE': TeamData(fullName: 'Cleveland Cavaliers',    primaryColor: Color(0xFF860038), accentColor: Color(0xFFFDBB30)),
    'DAL': TeamData(fullName: 'Dallas Mavericks',       primaryColor: Color(0xFF00538C), accentColor: Color(0xFF002B5E)),
    'DEN': TeamData(fullName: 'Denver Nuggets',         primaryColor: Color(0xFF0E2240), accentColor: Color(0xFFFEC524)),
    'DET': TeamData(fullName: 'Detroit Pistons',        primaryColor: Color(0xFFC8102E), accentColor: Color(0xFF006BB6)),
    'GSW': TeamData(fullName: 'Golden State Warriors',  primaryColor: Color(0xFF1D428A), accentColor: Color(0xFFFFC72C)),
    'HOU': TeamData(fullName: 'Houston Rockets',        primaryColor: Color(0xFFCE1141), accentColor: Color(0xFFC4CED4)),
    'IND': TeamData(fullName: 'Indiana Pacers',         primaryColor: Color(0xFF002D62), accentColor: Color(0xFFFDB927)),
    'LAC': TeamData(fullName: 'LA Clippers',            primaryColor: Color(0xFF1D428A), accentColor: Color(0xFFC8102E)),
    'LAL': TeamData(fullName: 'Los Angeles Lakers',     primaryColor: Color(0xFF552583), accentColor: Color(0xFFFDB927)),
    'MEM': TeamData(fullName: 'Memphis Grizzlies',      primaryColor: Color(0xFF5D76A9), accentColor: Color(0xFF12173F)),
    'MIA': TeamData(fullName: 'Miami Heat',             primaryColor: Color(0xFF98002E), accentColor: Color(0xFFF9A01B)),
    'MIL': TeamData(fullName: 'Milwaukee Bucks',        primaryColor: Color(0xFF00471B), accentColor: Color(0xFFEEE1C6)),
    'MIN': TeamData(fullName: 'Minnesota Timberwolves', primaryColor: Color(0xFF0C2340), accentColor: Color(0xFF236192)),
    'NOP': TeamData(fullName: 'New Orleans Pelicans',   primaryColor: Color(0xFF0C2340), accentColor: Color(0xFFC8102E)),
    'NYK': TeamData(fullName: 'New York Knicks',        primaryColor: Color(0xFF006BB6), accentColor: Color(0xFFF58426)),
    'OKC': TeamData(fullName: 'Oklahoma City Thunder',  primaryColor: Color(0xFF007AC1), accentColor: Color(0xFFEF3B24)),
    'ORL': TeamData(fullName: 'Orlando Magic',          primaryColor: Color(0xFF0077C0), accentColor: Color(0xFFC4CED4)),
    'PHI': TeamData(fullName: 'Philadelphia 76ers',     primaryColor: Color(0xFF006BB6), accentColor: Color(0xFFED174C)),
    'PHX': TeamData(fullName: 'Phoenix Suns',           primaryColor: Color(0xFF1D1160), accentColor: Color(0xFFE56020)),
    'POR': TeamData(fullName: 'Portland Trail Blazers', primaryColor: Color(0xFFE03A3E), accentColor: Color(0xFFBABEBE)),
    'SAC': TeamData(fullName: 'Sacramento Kings',       primaryColor: Color(0xFF5A2D81), accentColor: Color(0xFF63727A)),
    'SAS': TeamData(fullName: 'San Antonio Spurs',      primaryColor: Color(0xFFC4CED4), accentColor: Color(0xFF000000)),
    'TOR': TeamData(fullName: 'Toronto Raptors',        primaryColor: Color(0xFFCE1141), accentColor: Color(0xFF000000)),
    'UTA': TeamData(fullName: 'Utah Jazz',              primaryColor: Color(0xFF002B5C), accentColor: Color(0xFF00471B)),
    'WAS': TeamData(fullName: 'Washington Wizards',     primaryColor: Color(0xFF002B5C), accentColor: Color(0xFFE31837)),
  };

  static const Map<String, TeamData> nfl = {
    'ARI': TeamData(fullName: 'Arizona Cardinals',      primaryColor: Color(0xFF97233F), accentColor: Color(0xFFFFB612)),
    'ATL': TeamData(fullName: 'Atlanta Falcons',        primaryColor: Color(0xFFA71930), accentColor: Color(0xFFA5ACAF)),
    'BAL': TeamData(fullName: 'Baltimore Ravens',       primaryColor: Color(0xFF241773), accentColor: Color(0xFF9E7C0C)),
    'BUF': TeamData(fullName: 'Buffalo Bills',          primaryColor: Color(0xFF00338D), accentColor: Color(0xFFC60C30)),
    'CAR': TeamData(fullName: 'Carolina Panthers',      primaryColor: Color(0xFF0085CA), accentColor: Color(0xFF101820)),
    'CHI': TeamData(fullName: 'Chicago Bears',          primaryColor: Color(0xFF0B162A), accentColor: Color(0xFFC83803)),
    'CIN': TeamData(fullName: 'Cincinnati Bengals',     primaryColor: Color(0xFFFB4F14), accentColor: Color(0xFF000000)),
    'CLE': TeamData(fullName: 'Cleveland Browns',       primaryColor: Color(0xFF311D00), accentColor: Color(0xFFFF3C00)),
    'DAL': TeamData(fullName: 'Dallas Cowboys',         primaryColor: Color(0xFF003594), accentColor: Color(0xFF869397)),
    'DEN': TeamData(fullName: 'Denver Broncos',         primaryColor: Color(0xFF002244), accentColor: Color(0xFFFC4C02)),
    'DET': TeamData(fullName: 'Detroit Lions',          primaryColor: Color(0xFF0076B6), accentColor: Color(0xFFB0B7BC)),
    'GB':  TeamData(fullName: 'Green Bay Packers',      primaryColor: Color(0xFF203731), accentColor: Color(0xFFFFB612)),
    'HOU': TeamData(fullName: 'Houston Texans',         primaryColor: Color(0xFF03202F), accentColor: Color(0xFFFC4C02)),
    'IND': TeamData(fullName: 'Indianapolis Colts',     primaryColor: Color(0xFF002C5F), accentColor: Color(0xFFA2AAAD)),
    'JAX': TeamData(fullName: 'Jacksonville Jaguars',   primaryColor: Color(0xFF006778), accentColor: Color(0xFF9F792C)),
    'KC':  TeamData(fullName: 'Kansas City Chiefs',     primaryColor: Color(0xFFE31837), accentColor: Color(0xFFFFB81C)),
    'LAC': TeamData(fullName: 'Los Angeles Chargers',   primaryColor: Color(0xFF0080C6), accentColor: Color(0xFFFFC20E)),
    'LAR': TeamData(fullName: 'Los Angeles Rams',       primaryColor: Color(0xFF003594), accentColor: Color(0xFFFFD100)),
    'LV':  TeamData(fullName: 'Las Vegas Raiders',      primaryColor: Color(0xFF000000), accentColor: Color(0xFFA5ACAF)),
    'MIA': TeamData(fullName: 'Miami Dolphins',         primaryColor: Color(0xFF008E97), accentColor: Color(0xFFFC4C02)),
    'MIN': TeamData(fullName: 'Minnesota Vikings',      primaryColor: Color(0xFF4F2683), accentColor: Color(0xFFFFC62F)),
    'NE':  TeamData(fullName: 'New England Patriots',   primaryColor: Color(0xFF002244), accentColor: Color(0xFFC60C30)),
    'NO':  TeamData(fullName: 'New Orleans Saints',     primaryColor: Color(0xFF101820), accentColor: Color(0xFFD3BC8D)),
    'NYG': TeamData(fullName: 'New York Giants',        primaryColor: Color(0xFF0B2265), accentColor: Color(0xFFA71930)),
    'NYJ': TeamData(fullName: 'New York Jets',          primaryColor: Color(0xFF125740), accentColor: Color(0xFFFFFFFF)),
    'PHI': TeamData(fullName: 'Philadelphia Eagles',    primaryColor: Color(0xFF004C54), accentColor: Color(0xFFA5ACAF)),
    'PIT': TeamData(fullName: 'Pittsburgh Steelers',    primaryColor: Color(0xFFFFB612), accentColor: Color(0xFF101820)),
    'SEA': TeamData(fullName: 'Seattle Seahawks',       primaryColor: Color(0xFF002244), accentColor: Color(0xFF69BE28)),
    'SF':  TeamData(fullName: 'San Francisco 49ers',    primaryColor: Color(0xFFAA0000), accentColor: Color(0xFFB3995D)),
    'TB':  TeamData(fullName: 'Tampa Bay Buccaneers',   primaryColor: Color(0xFFD50A0A), accentColor: Color(0xFFFF7900)),
    'TEN': TeamData(fullName: 'Tennessee Titans',       primaryColor: Color(0xFF0C2340), accentColor: Color(0xFF4B92DB)),
    'WAS': TeamData(fullName: 'Washington Commanders',  primaryColor: Color(0xFF5A1414), accentColor: Color(0xFFFFB612)),
  };

  static TeamData? forTeam(String abbr, String league) {
    if (league == 'NBA') return nba[abbr];
    if (league == 'NFL') return nfl[abbr];
    return null;
  }

  static TeamData? anyLeague(String abbr) {
    return nba[abbr] ?? nfl[abbr];
  }

  static List<String> nbaAbbrs() => nba.keys.toList()..sort();
  static List<String> nflAbbrs() => nfl.keys.toList()..sort();
}

File: web/src/lib/teamConfig.ts
export interface TeamData {
  fullName: string
  primaryColor: string
  accentColor: string
}

export const NBA_TEAMS: Record<string, TeamData> = {
  ATL: { fullName: 'Atlanta Hawks',          primaryColor: '#E03A3E', accentColor: '#C1D32F' },
  BOS: { fullName: 'Boston Celtics',          primaryColor: '#007A33', accentColor: '#BA9653' },
  BKN: { fullName: 'Brooklyn Nets',           primaryColor: '#000000', accentColor: '#FFFFFF' },
  CHA: { fullName: 'Charlotte Hornets',       primaryColor: '#1D1160', accentColor: '#00788C' },
  CHI: { fullName: 'Chicago Bulls',           primaryColor: '#CE1141', accentColor: '#000000' },
  CLE: { fullName: 'Cleveland Cavaliers',     primaryColor: '#860038', accentColor: '#FDBB30' },
  DAL: { fullName: 'Dallas Mavericks',        primaryColor: '#00538C', accentColor: '#002B5E' },
  DEN: { fullName: 'Denver Nuggets',          primaryColor: '#0E2240', accentColor: '#FEC524' },
  DET: { fullName: 'Detroit Pistons',         primaryColor: '#C8102E', accentColor: '#006BB6' },
  GSW: { fullName: 'Golden State Warriors',   primaryColor: '#1D428A', accentColor: '#FFC72C' },
  HOU: { fullName: 'Houston Rockets',         primaryColor: '#CE1141', accentColor: '#C4CED4' },
  IND: { fullName: 'Indiana Pacers',          primaryColor: '#002D62', accentColor: '#FDBB30' },
  LAC: { fullName: 'LA Clippers',             primaryColor: '#C8102E', accentColor: '#1D428A' },
  LAL: { fullName: 'Los Angeles Lakers',      primaryColor: '#552583', accentColor: '#FDB927' },
  MEM: { fullName: 'Memphis Grizzlies',       primaryColor: '#5D76A9', accentColor: '#12173F' },
  MIA: { fullName: 'Miami Heat',              primaryColor: '#98002E', accentColor: '#F9A01B' },
  MIL: { fullName: 'Milwaukee Bucks',         primaryColor: '#00471B', accentColor: '#EEE1C6' },
  MIN: { fullName: 'Minnesota Timberwolves',  primaryColor: '#0C2340', accentColor: '#236192' },
  NOP: { fullName: 'New Orleans Pelicans',    primaryColor: '#0C2340', accentColor: '#C8102E' },
  NYK: { fullName: 'New York Knicks',         primaryColor: '#006BB6', accentColor: '#F58426' },
  OKC: { fullName: 'Oklahoma City Thunder',   primaryColor: '#007AC1', accentColor: '#EF3B24' },
  ORL: { fullName: 'Orlando Magic',           primaryColor: '#0077C0', accentColor: '#C4CED4' },
  PHI: { fullName: 'Philadelphia 76ers',      primaryColor: '#006BB6', accentColor: '#ED174C' },
  PHX: { fullName: 'Phoenix Suns',            primaryColor: '#1D1160', accentColor: '#E56020' },
  POR: { fullName: 'Portland Trail Blazers',  primaryColor: '#E03A3E', accentColor: '#000000' },
  SAC: { fullName: 'Sacramento Kings',        primaryColor: '#5A2D81', accentColor: '#63727A' },
  SAS: { fullName: 'San Antonio Spurs',       primaryColor: '#C4CED4', accentColor: '#000000' },
  TOR: { fullName: 'Toronto Raptors',         primaryColor: '#CE1141', accentColor: '#000000' },
  UTA: { fullName: 'Utah Jazz',               primaryColor: '#002B5C', accentColor: '#00471B' },
  WAS: { fullName: 'Washington Wizards',      primaryColor: '#002B5C', accentColor: '#E31837' },
}

export const NFL_TEAMS: Record<string, TeamData> = {
  ARI: { fullName: 'Arizona Cardinals',       primaryColor: '#97233F', accentColor: '#000000' },
  ATL: { fullName: 'Atlanta Falcons',         primaryColor: '#A71930', accentColor: '#000000' },
  BAL: { fullName: 'Baltimore Ravens',        primaryColor: '#241773', accentColor: '#9E7C0C' },
  BUF: { fullName: 'Buffalo Bills',           primaryColor: '#00338D', accentColor: '#C60C30' },
  CAR: { fullName: 'Carolina Panthers',       primaryColor: '#0085CA', accentColor: '#101820' },
  CHI: { fullName: 'Chicago Bears',           primaryColor: '#0B162A', accentColor: '#C83803' },
  CIN: { fullName: 'Cincinnati Bengals',      primaryColor: '#FB4F14', accentColor: '#000000' },
  CLE: { fullName: 'Cleveland Browns',        primaryColor: '#311D00', accentColor: '#FF3C00' },
  DAL: { fullName: 'Dallas Cowboys',          primaryColor: '#003594', accentColor: '#869397' },
  DEN: { fullName: 'Denver Broncos',          primaryColor: '#FB4F14', accentColor: '#002244' },
  DET: { fullName: 'Detroit Lions',           primaryColor: '#0076B6', accentColor: '#B0B7BC' },
  GB:  { fullName: 'Green Bay Packers',       primaryColor: '#203731', accentColor: '#FFB612' },
  HOU: { fullName: 'Houston Texans',          primaryColor: '#03202F', accentColor: '#A71930' },
  IND: { fullName: 'Indianapolis Colts',      primaryColor: '#002C5F', accentColor: '#A2AAAD' },
  JAX: { fullName: 'Jacksonville Jaguars',    primaryColor: '#101820', accentColor: '#D7A22A' },
  KC:  { fullName: 'Kansas City Chiefs',      primaryColor: '#E31837', accentColor: '#FFB81C' },
  LAC: { fullName: 'Los Angeles Chargers',    primaryColor: '#0080C6', accentColor: '#FFC20E' },
  LAR: { fullName: 'Los Angeles Rams',        primaryColor: '#003594', accentColor: '#FFA300' },
  LV:  { fullName: 'Las Vegas Raiders',       primaryColor: '#000000', accentColor: '#A5ACAF' },
  MIA: { fullName: 'Miami Dolphins',          primaryColor: '#008E97', accentColor: '#FC4C02' },
  MIN: { fullName: 'Minnesota Vikings',       primaryColor: '#4F2683', accentColor: '#FFC62F' },
  NE:  { fullName: 'New England Patriots',    primaryColor: '#002244', accentColor: '#C60C30' },
  NO:  { fullName: 'New Orleans Saints',      primaryColor: '#D3BC8D', accentColor: '#101820' },
  NYG: { fullName: 'New York Giants',         primaryColor: '#0B2265', accentColor: '#A71930' },
  NYJ: { fullName: 'New York Jets',           primaryColor: '#125740', accentColor: '#000000' },
  PHI: { fullName: 'Philadelphia Eagles',     primaryColor: '#004C54', accentColor: '#A5ACAF' },
  PIT: { fullName: 'Pittsburgh Steelers',     primaryColor: '#FFB612', accentColor: '#101820' },
  SEA: { fullName: 'Seattle Seahawks',        primaryColor: '#002244', accentColor: '#69BE28' },
  SF:  { fullName: 'San Francisco 49ers',     primaryColor: '#AA0000', accentColor: '#B3995D' },
  TB:  { fullName: 'Tampa Bay Buccaneers',    primaryColor: '#D50A0A', accentColor: '#FF7900' },
  TEN: { fullName: 'Tennessee Titans',        primaryColor: '#0C2340', accentColor: '#4B92DB' },
  WAS: { fullName: 'Washington Commanders',   primaryColor: '#5A1414', accentColor: '#FFB612' },
}

export function getTeam(abbr: string, league: string): TeamData | undefined {
  if (league === 'NBA') return NBA_TEAMS[abbr]
  if (league === 'NFL') return NFL_TEAMS[abbr]
  return NBA_TEAMS[abbr] ?? NFL_TEAMS[abbr]
}

export function getTeamColor(abbr: string, league: string): string {
  return getTeam(abbr, league)?.primaryColor ?? '#6B7280'
}

export function getTeamName(abbr: string, league: string): string {
  return getTeam(abbr, league)?.fullName ?? abbr
}

export const ALL_LEAGUES = ['NBA', 'NFL', 'NCAAB'] as const
export type League = typeof ALL_LEAGUES[number]

export function getTeamsByLeague(league: 'NBA' | 'NFL'): Record<string, TeamData> {
  return league === 'NBA' ? NBA_TEAMS : NFL_TEAMS
}
